[Build Error] Cannot build gettext:target due to build host (gentoo) picking up system xml2

<!-- Before making a bug report please visit https//forum.libreelec.tv where -->
<!-- a large community exists to help and support you. -->

<!-- Opening a bug report without triaging the issue in the forum first will -->
<!-- probably result in the report being closed, with an instruction to visit -->
<!-- the forum for support. -->

### Describe the bug
<!-- A clear description of what the bug is. -->

Building current master (8fcc5879b2cdc443a28098485845048286fbc3cf) causes `gettext:target` to fail with:

`libtool: link: /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-gcc -fanalyzer -fstrict-flex-arrays -Wall -Warith-conversion -Wcast-align=strict -Wdate-time -Wdouble-promotion -Wduplicated-cond -Wextra -Wflex-array-member-not-at-end -Wformat-signedness -Winit-self -Winvalid-pch -Wlogical-op -Wmissing-include-dirs -Wmissing-variable-declarations -Wnested-externs -Wnull-dereference -Wopenmp-simd -Woverlength-strings -Wpacked -Wpointer-arith -Wstrict-flex-arrays -Wstrict-overflow -Wsuggest-attribute=format -Wsync-nand -Wtrampolines -Wuninitialized -Wunknown-pragmas -Wunsafe-loop-optimizations -Wvariadic-macros -Wvector-operation-performance -Wwrite-strings -Warray-bounds=2 -Wattribute-alias=2 -Wbidi-chars=any,ucn -Wformat-overflow=2 -Wformat=2 -Wformat-truncation=2 -Wimplicit-fallthrough=5 -Wshift-overflow=2 -Wuse-after-free=3 -Wunused-const-variable=2 -Wvla-larger-than=4031 -Wzero-as-null-pointer-constant -Wno-analyzer-malloc-leak -Wno-empty-body -Wno-analyzer-double-fclose -Wno-analyzer-double-free -Wno-analyzer-fd-double-close -Wno-analyzer-null-argument -Wno-analyzer-null-dereference -Wno-analyzer-use-of-uninitialized-value -Wno-cast-align -Wno-format-nonliteral -Wno-sign-compare -Wno-type-limits -Wno-unused-parameter -Wno-clobbered -Wshadow=local -march=x86-64 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -fuse-ld=gold -o .libs/msgcmp msgcmp-msgcmp.o msgcmp-msgl-fsearch.o  -Wl,--as-needed ./.libs/libgettextsrc.so -L/usr/lib64 /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-1.0/.x86_64-libreelec-linux-gnu/gettext-tools/gnulib-lib/.libs/libgettextlib.so -lattr -lxml2 /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-1.0/.x86_64-libreelec-linux-gnu/libtextstyle/lib/.libs/libtextstyle.so -lc -lm `

`/home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-1.0/.x86_64-libreelec-linux-gnu/gettext-tools/gnulib-lib/.libs/libgettextlib.so: error: undefined reference to 'memset_explicit'
collect2: error: ld returned 1 exit status`

### How to reproduce
Steps to reproduce the behavior:
1. Check out/pull 8fcc5879b2cdc443a28098485845048286fbc3cf
2. `PROJECT=Generic ARCH=x86_64 make distclean`
3. `PROJECT=Generic ARCH=x86_64 make image`
4. Build fails for gettext

### Information
 - LibreELEC Version: 8fcc5879b2cdc443a28098485845048286fbc3cf
 - Hardware Platform: Generic x86_64

### Log file
<!-- Add debug log files (https://libreelec.wiki/how_to/provide_logfile) that we can search for errors. -->

[124.log](https://github.com/user-attachments/files/26575292/124.log)

### Context
<!-- Add any context about the problem here. -->

- The C library for the system image appears to contain a definition of `memset_explicit`:
```
$ nm -D build.LibreELEC-Generic.x86_64-13.0-devel/image/system/usr/lib/libc.so.6|grep memset_explicit
000000000009e720 T memset_explicit@@GLIBC_2.43
000000000010f0f0 T __memset_explicit_chk@@GLIBC_2.43
```

- The reference to `memset_explicit` in `libgettextlib.so` that's attempted to be linked is not decorated with the glibc version:
```
nm -D build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-1.0/.x86_64-libreelec-linux-gnu/gettext-tools/gnulib-lib/.libs/libgettextlib.so | grep memset_explicit
                 U memset_explicit
```

- I've tried passing `HAVE_MEMSET_EXPLICIT=0` (and also `REPLACE_MEMSET_EXPLICIT=1`) to `gettext`'s `configure` (in its `package.mk`, but no change

- `/usr/lib64/libc.so.6` on my host system does **not** define `memset_explicit`
  - I don't know `libtool` in a cross build context well enough: (how) would the given library directory of `-L/usr/lib64` (in the linker line above) know to assume `.../build.LibreELEC-Generic.x86_64-13.0-devel/image/system/` as the root? I kinda would have expected to find the full path to my cross-built system image there. **Also, the `lib64` reeks of my host system, rather than the LE system image which (only) has `/usr/lib` (no `lib64`)!**

- Trying to revert (only) 75fc7b25ad0acd248495883ceac37a5ecc4ede59 (taking `gettext` from 1.0 back to 0.26, its most recent version bump) does not straightforwardly work, I get errors around `wmemchar` definitions:
```
...
libtool: compile:  /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I/home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-0.26/gettext-runtime/intl/gnulib-lib -I.. -fanalyzer -fstrict-flex-arrays -Wall -Warith-conversion -Wcast-align=strict -Wdate-time -Wdouble-promotion -Wduplicated-cond -Wextra -Wformat-signedness -Wflex-array-member-not-at-end -Winit-self -Winvalid-pch -Wlogical-op -Wmissing-include-dirs -Wmissing-variable-declarations -Wnested-externs -Wnull-dereference -Wopenmp-simd -Woverlength-strings -Wpacked -Wpointer-arith -Wstrict-flex-arrays -Wstrict-overflow -Wsuggest-attribute=format -Wsuggest-final-methods -Wsuggest-final-types -Wsync-nand -Wtrampolines -Wuninitialized -Wunknown-pragmas -Wunsafe-loop-optimizations -Wvariadic-macros -Wvector-operation-performance -Wwrite-strings -Warray-bounds=2 -Wattribute-alias=2 -Wbidi-chars=any,ucn -Wformat-overflow=2 -Wformat=2 -Wformat-truncation=2 -Wimplicit-fallthrough=5 -Wshift-overflow=2 -Wuse-after-free=3 -Wunused-const-variable=2 -Wvla-larger-than=4031 -Wno-analyzer-malloc-leak -Wno-empty-body -Wno-analyzer-double-fclose -Wno-analyzer-double-free -Wno-analyzer-fd-double-close -Wno-analyzer-null-argument -Wno-analyzer-null-dereference -Wno-analyzer-use-of-uninitialized-value -Wno-cast-align -Wno-format-nonliteral -Wno-sign-compare -Wno-type-limits -Wno-unused-parameter -Wno-clobbered -Wshadow=local -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -Wno-unused-const-variable -Wno-unsuffixed-float-constants -Wno-error -march=x86-64 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -MT libgnu_la-c32isdigit.lo -MD -MP -MF .deps/libgnu_la-c32isdigit.Tpo -c /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-0.26/gettext-runtime/intl/gnulib-lib/c32isdigit.c  -fPIC -DPIC -o .libs/libgnu_la-c32isdigit.o
In file included from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/features.h:540,
                 from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/uchar.h:26,
                 from ./uchar.h:45,
                 from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-0.26/gettext-runtime/intl/gnulib-lib/c32isblank.c:21:
./wchar.h:856:23: error: expected identifier or '(' before '_Generic'
  856 | _GL_EXTERN_C wchar_t *wmemchr (const wchar_t *__s, wchar_t __wc, size_t __n)
      |                       ^~~~~~~
make[9]: *** [Makefile:2847: libgnu_la-c32isblank.lo] Error 1
make[9]: *** Waiting for unfinished jobs....
In file included from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/features.h:540,
                 from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/uchar.h:26,
                 from ./uchar.h:45,
                 from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-0.26/gettext-runtime/intl/gnulib-lib/c32iscntrl.c:21:
./wchar.h:856:23: error: expected identifier or '(' before '_Generic'
  856 | _GL_EXTERN_C wchar_t *wmemchr (const wchar_t *__s, wchar_t __wc, size_t __n)
      |                       ^~~~~~~
make[9]: *** [Makefile:2854: libgnu_la-c32iscntrl.lo] Error 1
In file included from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/features.h:540,
                 from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/uchar.h:26,
                 from ./uchar.h:45,
                 from /home/jmbreuer/Developer/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/build/gettext-0.26/gettext-runtime/intl/gnulib-lib/c32isalpha.c:21:
./wchar.h:856:23: error: expected identifier or '(' before '_Generic'
...
```

--- a/packages/devel/gettext/package.mk
+++ b/packages/devel/gettext/package.mk
@@ -9,7 +9,7 @@ PKG_LICENSE="GPL-3.0-or-later AND LGPL-2.1-or-later"
 PKG_SITE="https://www.gnu.org/s/gettext/"
 PKG_URL="https://ftp.gnu.org/pub/gnu/gettext/${PKG_NAME}-${PKG_VERSION}.tar.xz"
 PKG_DEPENDS_HOST="make:host"
-PKG_DEPENDS_TARGET="autotools:host make:host gcc:host"
+PKG_DEPENDS_TARGET="autotools:host make:host gcc:host libxml2"
 PKG_LONGDESC="A program internationalization library and tools."
 PKG_BUILD_FLAGS="+local-cc"
 
@@ -27,6 +27,9 @@ PKG_CONFIGURE_OPTS_HOST="${PKG_CONFIGURE_OPTS_COMMON} \
                          --without-emacs"
 
 PKG_CONFIGURE_OPTS_TARGET="${PKG_CONFIGURE_OPTS_COMMON} \
+                           --disable-curses \
+                           --disable-xattr \
+                           --with-libxml2-prefix=${SYSROOT_PREFIX}/usr \
                            --with-sysroot=yes"
 
 post_configure_target() {

